### PR TITLE
[WIP] Improve network monitor

### PR DESCRIPTION
**Pull request by alexisbatyk**

### CODE REVIEW

File: crates/core/src/client_events.rs [hunk 1]
```diff
@@ -377,7 +377,25 @@ pub(crate) mod test {
...
+                            }
+
+                            let key = contract.key();
+                            let request = ContractRequest::Put {
+                                contract: contract.clone(),
+                                state: WrappedState::new(self.random_byte_vec()),
+                                related_contracts: RelatedContracts::new(),
+                            };
+
+                            println!("sending put to an existing contract");
```
Comment (al8n) [inline crates/core/src/client_events.rs:393]: left-over?

File: crates/fdev/src/network_metrics_server.rs [hunk 2]
```diff
@@ -127,7 +130,30 @@ async fn push_interface(ws: WebSocket, state: Arc<ServerState>) -> anyhow::Resul
...
+                            format!("{decoding_error}"),
+                        ))))
+                        .await?;
+                    }
+                }
+                match ContractChange::try_decode_fbs(&msg) {
+                    Ok(ContractChange::PutFailure(err)) => {
+                        // FIXME: handle put failure
+                        tracing::error!(error = "Failed to put contract");
+                    }
```
Comment (al8n) [inline crates/fdev/src/network_metrics_server.rs:153]: It seems that we do not really handle `PutFailure` here.

Comment (iduartgomez) [reply crates/fdev/src/network_metrics_server.rs:153]: Yes that's not done yet, but is ok, we can come back to this later if we need to.

File: stdlib
Comment (iduartgomez) [inline stdlib:1]: @al8n can you check this out, we are changing to what commit on sodlib we are pointing to, let's make sure this changes are in  `main` in stdlib and just point to head in sodlib

**Pull request merged**